graph/api/api.go

```diff
@@ -93,11 +95,107 @@ func generateGraph(trafficMap graph.TrafficMap, o graph.Options) (int, interface
 	var vendorConfig interface{}
 	switch o.ConfigVendor {
 	case graph.VendorCytoscape:
-		vendorConfig = cytoscape.NewConfig(trafficMap, o.ConfigOptions)
+		vc := cytoscape.NewConfig(trafficMap, o.ConfigOptions)
+		if o.ConfigOptions.IncludeHealth {
+			err := collectGraphHealth(&vc, o.ConfigOptions, bs, ctx)
+			if err != nil {
+				graph.Error(fmt.Sprintf("Error collecting graph health: %s", err.Error()))
+			}
+		}
+		vendorConfig = &vc
 	default:
 		graph.Error(fmt.Sprintf("ConfigVendor [%s] not supported", o.ConfigVendor))
 	}
 
 	log.Tracef("Done generating config for [%s] graph", o.ConfigVendor)
 	return http.StatusOK, vendorConfig
 }
+
+func collectGraphHealth(vendorConfig *cytoscape.Config, o graph.ConfigOptions, bs *business.Layer, ctx context.Context) error {
+	healthReqs := make(map[string]map[string][]*cytoscape.NodeWrapper)
+
+	// Limit health fetches to only the necessary namespaces for the necessary types
+	for _, node := range vendorConfig.Elements.Nodes {
+		if node.Data.IsInaccessible {
+			continue
+		}
+
+		kind := ""
+		namespace := node.Data.Namespace
+		nodeType := node.Data.NodeType
+		workloadOk := len(node.Data.Workload) != 0 && node.Data.Workload != graph.Unknown
+
+		useWorkloadHealth := nodeType == graph.NodeTypeWorkload || (nodeType == graph.NodeTypeApp && workloadOk)
+		if useWorkloadHealth {
+			kind = "workload"
+		} else {
+			switch nodeType {
+			case graph.NodeTypeApp:
+				kind = "app"
+			case graph.NodeTypeBox:
+				if node.Data.IsBox == graph.BoxByApp {
+					kind = "app"
+				}
+			case graph.NodeTypeService:
+				kind = "service"
+			}
+		}
+
+		if len(kind) != 0 {
+			if _, nsOk := healthReqs[namespace]; !nsOk {
+				healthReqs[namespace] = make(map[string][]*cytoscape.NodeWrapper)
+			}
+
+			healthReqs[namespace][kind] = append(healthReqs[namespace][kind], node)
+		}
+	}
+
+	// Execute health fetches and attach retrieved health data to nodes
+	for namespace, kinds := range healthReqs {
+		for kind, nodeWrappers := range kinds {
+			switch kind {
+			case "app":
+				health, err := bs.Health.GetNamespaceAppHealth(ctx, namespace, o.RawDuration, time.Unix(o.QueryTime, 0))
+				if err != nil {
+					return err
+				}
+				for _, nodeWrap := range nodeWrappers {
+					if h, ok := health[nodeWrap.Data.App]; ok {
+						nodeWrap.Data.HealthData = h
+					} else {
+						nodeWrap.Data.HealthData = []int{}
```

Comment:
Why `[]int{}`?

Reply:
:+1: I don't think it matters what type it is during serialization if the array is empty

Reply:
If you go with the other comments and always supply health for graph components, removing the `includeHealth` option, then perhaps you can just use omitempty when there is nothing to report (or maybe even you can then assume healthy). 
Otherise the health data would be provided.   Just a thought.